### PR TITLE
Pagination doesn't work in resource search

### PR DESCRIPTION
**Pull request by jotare**
Story details: 
[sc-4011] https://app.shortcut.com/flaps/story/4011
[sc-4023] https://app.shortcut.com/flaps/story/4023

Added same pagination strategy we are doing in KB search

Added more exhaustive tests for pagination

### CODE REVIEW

File: nucliadb/nucliadb/search/search/query.py [hunk 1]
```diff
@@ -178,6 +178,11 @@ async def paragraph_query_to_pb(
     request = ParagraphSearchRequest()
     request.reload = reload
     request.with_duplicates = with_duplicates
+
+    # We need to ask for all and cut later
+    request.page_number = 0
+    request.result_per_page = page_number * page_size + page_size
+
     if SearchOptions.PARAGRAPH in features:
```
Comment (lferran) [inline nucliadb/nucliadb/search/search/query.py:186]: Does it make sense to query the resource search and not have PARAGRAPH as requested feature?

File: nucliadb/nucliadb/search/api/v1/resource.py [hunk 2]
```diff
@@ -86,8 +86,6 @@ async def search(
     range_modification_end: Optional[datetime] = None,
     features: List[SearchOptions] = [
         SearchOptions.PARAGRAPH,
-        SearchOptions.VECTOR,
```
Comment (lferran) [inline nucliadb/nucliadb/search/api/v1/resource.py:89]: Like this, if the user requests vectors or relations we will simply ignore it.
Have you considered returning a 4xx instead?

File: nucliadb/nucliadb/search/search/query.py (hunk 1 above)
Comment (jotare) [reply nucliadb/nucliadb/search/search/query.py:186]: Yes, you can search by dates for example (although it's not implemented either (I'll add it now))

File: nucliadb/nucliadb/search/api/v1/resource.py (hunk 2 above)
Comment (jotare) [reply nucliadb/nucliadb/search/api/v1/resource.py:89]: I've just removed the option, as we don't want to provide it

File: nucliadb/nucliadb/search/api/v1/resource.py [hunk 3]
```diff
@@ -101,6 +99,12 @@ async def search(
     debug: bool = Query(False),
     shards: List[str] = Query(default=[]),
 ) -> ResourceSearchResults:
+    if not all(map(lambda feature: feature == SearchOptions.PARAGRAPH, features)):
+        raise HTTPException(
+            status_code=422,
+            detail=f"Invalid search option. Only {SearchOptions.PARAGRAPH} is valid",
+        )
```
Comment (lferran) [inline nucliadb/nucliadb/search/api/v1/resource.py:106]: If there is only one option, why providing the parameter at all?
I am in for removing the param and returning paragraphs always. Then we don't need this extra logic.
If we happen to offer more features in the future we can add it back easily.
WDYT?

Comment (jotare) [reply nucliadb/nucliadb/search/api/v1/resource.py:106]: Seems fine to me!

**Review by lferran — APPROVED** [review]

**Pull request merged**